### ISSUE TITLE
Support for datadog tracing

### ISSUE THREAD

**Issue by devcsrj** [frain-dev/convoy#1816]
https://docs.datadoghq.com/

The PR https://github.com/frain-dev/convoy/pull/264 added support for New Relic. However, the interface is tightly coupled to NewRelic

https://github.com/frain-dev/convoy/blob/cafc57e40d2c1ab2099a8698758fda6562181a9d/tracer/tracer.go#L13-L19

This makes it not trivial to add support other instrumentation libraries

**Comment by subomi** [comment]
Hey @devcsrj, OTel support just landed on main. A release will go out next week.

**Comment by subomi** [comment]
Hey @devcsrj , this has been released in [v24.1.1](https://github.com/frain-dev/convoy/releases/tag/v24.1.1)

**Comment by devcsrj** [comment]
Nice! Thanks @subomi ! We'll try it out

**Comment by subomi** [comment]
Yeah, I'm sure there are gaps, so don't hesitate to report issues; I'm looking to iterate on it this month! @devcsrj
